docs/001_develop/03_client-capabilities/023_custom-components/05_directives

```diff
@@ -0,0 +1,291 @@
+# DIRECTIVES
+
+## Ref
+
+The ref directive lets you directly reference a specific DOM element within your component's template, enabling direct interaction with that element.
+
+Here’s how to use ref in <my-button> to reference the button element, making it accessible for direct manipulation in code.
+
+```shell
+import {customElement, GenesisElement, html, ref } from '@genesislcap/web-core';
+
+@customElement({
+    name: "my-button",
+    template: html`
+        <button ${ref("buttonElement")}>Click Me</button>
+    `,
+})
+export class MyButton extends GenesisElement {
+    buttonElement: HTMLButtonElement;
+
+    connectedCallback() {
+        super.connectedCallback();
+        console.log("Button text:", this.buttonElement.textContent);
+    }
+}
+
+```
+
+- `ref("buttonElement")`: The ref directive assigns the `<button>` element to the `buttonElement` property on the `<my-button>` component instance.
+
+- In `connectedCallback`, `this.buttonElement` is a direct reference to the `<button>` DOM node, allowing us to access properties like `textContent`. Here, we log the button’s text to the console.
+
+:::note
+
+Even though the `<button>` element is part of our `<my-button>` component’s template, you can’t directly access DOM elements inside the template without a reference. The `ref` directive provides this direct access, allowing you to manipulate or retrieve information from a specific DOM element.
+
+:::
+
+### Syntax breakdown
+
+```shell
+ref(propertyName);
+```
+- Property Name: The name of the property that will store the DOM reference in the component. The element with ref(propertyName) will be assigned to this property.
+
+## Slotted
```

Comment:
This is all really good. It has me wondering if we should introduce the concept of `slot` in the html earlier on though in addition to this, I am unsure just introducing it here makes it clear what the use case is.